addons/website/models/ir_http.py

```diff
@@ -371,8 +371,8 @@ def generate(self, query=None, args={}):
         query = query and query.startswith('website.') and query[8:] or query
         if query:
             domain += [('key', 'like', query)]
-        website_id = request.context.get('website_id') or request.env['website'].search([], limit=1).id
-        domain += ['|', ('website_id', '=', website_id), ('website_id', '=', False)]
+        website = request.env['website'].get_current_website()
```

Comment:
Even if it seems right to me. Can you explain me the reason of this modification ? Is it in the purpose of the task ?